runtime/win32.c

```diff
@@ -1019,3 +1019,26 @@ int caml_num_rows_fd(int fd)
 {
   return -1;
 }
+
+/* UCRT clock function returns wall-clock time */
+CAMLexport clock_t caml_win32_clock(void)
+{
+  FILETIME c, e, stime, utime;
+  ULARGE_INTEGER tmp;
+  ULONGLONG total, clocks_per_sec;
+
+  if (!(GetProcessTimes(GetCurrentProcess(), &c, &e, &stime, &utime))) {
+    return (clock_t)(-1);
+  }
+
+  tmp.u.LowPart = stime.dwLowDateTime;
+  tmp.u.HighPart = stime.dwHighDateTime;
+  total = tmp.QuadPart;
+  tmp.u.LowPart = utime.dwLowDateTime;
+  tmp.u.HighPart = utime.dwHighDateTime;
+  total += tmp.QuadPart;
+
+  /* total in 100-nanosecond intervals (1e7 / CLOCKS_PER_SEC) */
+  clocks_per_sec = INT64_LITERAL(10000000U) / (ULONGLONG)CLOCKS_PER_SEC;
```

Comment:
Naive question: https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-getprocesstimes says
> All times are expressed using FILETIME data structures. Such a structure contains two 32-bit values that combine to form a 64-bit count of 100-nanosecond time units.

which I interpret to mean that the units of `total` is 100-nanoseconds (`1e-7` seconds). I would then expect `clocks_per_sec` to be simply `INT64_LITERAL(10000000U)`.

I don't understand what `CLOCKS_PER_SEC` is doing here. What am I missing?

Reply:
I've found that there's no such thing as naïve where time is concerned!

`CLOCKS_PER_SEC` is the number of `clock_t`s in a second, so the return value of `GetProcessTimes` needs to be divided by (1e7 / CLOCKS_PER_SEC) to convert it... assuming I haven't messed up my arithmetic (which is highly possible!)

Reply:
Ah, I see it now, this function is supposed to return `clock_t`'s, not seconds. Thanks!